### ISSUE TITLE
Clarification on the KNU-DTI Model Versions

### ISSUE THREAD

**Issue by zt20001126** [DBpackage/KNU_DTA#9]
Dear Heo,

I hope this email finds you well. While reproducing the KNU-DTI model from your paper, I noticed that the table titled "Comparison results of the proposed model and baselines on the BindingDB classification" mentions three versions of the model: KNU-DTI (Base), KNU-DTI (Full), and KNU-DTI (Full, ensemble3). I would like to further understand the specific meanings and differences between these versions.

From my reading of the paper, it seems that these three models differ in structure and functionality. Could you please clarify:

What exactly does KNU-DTI (Base) entail, and what are its core components?

How does KNU-DTI (Full) differ from KNU-DTI (Base), and what additional modules or features are included in KNU-DTI (Full) and KNU-DTI (Full, ensemble3)?

Could you provide a more detailed explanation of how the ensemble learning method is applied in KNU-DTI (Full, ensemble3)?

Thank you very much for your time and assistance. I look forward to your response.

Best regards,
Zhang

**Comment by zt20001126** [comment]
In the training logs, I see several outputs (Model 0, Model 1, Model 2, and Ensemble model), but I am unsure how these correspond to the KNU-DTI (Base), KNU-DTI (Full), and KNU-DTI (Full, ensemble3) models described in your paper. Could you kindly clarify which model corresponds to which result?

Fold 0
Model 0 best validation auc = 0.915820 on epoch 51
Model 0 test auc = 0.912592
Model 0 test prc-auc = 0.933996
Model 0 test accuracy = 0.840814
Model 1 best validation auc = 0.916211 on epoch 80
Model 1 test auc = 0.913249
Model 1 test prc-auc = 0.933626
Model 1 test accuracy = 0.841512
Model 2 best validation auc = 0.917121 on epoch 37
Model 2 test auc = 0.914805
Model 2 test prc-auc = 0.935024
Model 2 test accuracy = 0.839457
Ensemble test auc = 0.924149
Ensemble test prc-auc = 0.942856
Ensemble test accuracy = 0.852636
1-fold cross validation
	Seed 42 ==> test auc = 0.924149
	Seed 42 ==> test prc-auc = 0.942856
	Seed 42 ==> test accuracy = 0.852636
Overall test auc = 0.924149 +/- 0.000000
Overall test Cindex = 0.924149 +/- 0.000000 
Overall test prc-auc = 0.942856 +/- 0.000000
Overall test Cindex = 0.924149 +/- 0.000000 
Overall test accuracy = 0.852636 +/- 0.000000
Overall test Cindex = 0.924149 +/- 0.000000 
Elapsed time = 16:05:56

Thank you for your assistance, and I look forward to your response.

Best regards,
Zhang

**Comment by DBpackage** [comment]
Dear Zhang,

Thank you for your continued interest in our model.
I would like to mention that due to our recent busy schedules, there might be some delays in our responses.

KNU-DTI utilizes the following five types of representations:

(1) SPS information using GCNN
(2) Protein information based on Transformer
(3) Protein information based on D-MPNN
(4) Compound information based on Transformer
(5) ECFP information processed through MLP

The "Baseline" model referred to in the paper only uses the Transformer-based modules for proteins (2) and compounds (4).

To implement the Baseline model, you simply need to remove the options for SPS, D-MPNN, and ECFP in the train and test execution commands: --sps_cnn, --D_MPNN, and --ECFP.
If these three options are not provided, they are automatically set to False internally, and the model will be trained using only the Transformer-based modules, as intended for the baseline setting.

Conversely, the "Full" model uses all modules.

"Ensemble3" refers to a soft-voting ensemble.
It combines the outputs (probabilities for classification or logits for regression) of three independently and randomly initialized models to achieve the best performance.
(Note: The order of training and validation data is not altered; only the model parameter initialization is randomized.)

However, using ensemble models can make fair performance comparisons difficult.
Therefore, if you want to evaluate the performance of a single model without ensembling, you can set --ensemble_size to 1 instead of 3.

The reason you see results for model 0, 1, and 2 is because --ensemble_size 3 was used.
If you set --ensemble_size 1, only the result from model 0 will be returned.

+Additionally, please note that the baseline model is not appropriate for use as a comparison model.
Moreover, inference scripts would also require modification to work properly with the baseline model.
Therefore, we recommend using the training and inference codes provided in the README, which are based on the Full model.

I hope this clarifies your questions.
Please feel free to reach out if there are any further points that need clarification.

Thank you again.

Kind regards,
Ryong

**Comment by zt20001126** [comment]
Dear Ryong,

Thank you very much for taking the time to provide such a detailed and clear response to my questions! Your explanation has greatly helped me understand the implementation and details of the KNU-DTI model, which will be very beneficial for my future research.

I truly appreciate your patience and support! Wishing you all the best in your work, good health, and continued success.

Best regards,

Zhang

**Comment by zt20001126** [comment]
To ensure that my understanding is correct, could you please confirm whether Model 0, Model 1, and Model 2 are independently trained instances of the Full model with different random initializations?

**Comment by zt20001126** [comment]
To ensure that my understanding is correct, I would like to ask,
are Model 0, Model 1, and Model 2 independently trained instances of the Full model obtained through different random initializations?

**Comment by DBpackage** [comment]
Yes, you understood exactly 👍

**Comment by zt20001126** [comment]
I am deeply grateful for your time and detailed response!
